pkg/contextkeys/contextkeys.go

```diff
@@ -0,0 +1,41 @@
+package contextkeys
+
+import "context"
+
+// ContextKeys gets and sets context values from a context.
+type contextKeys struct {
+	accountID struct{}
+}
+
+var (
+	keys = contextKeys{
+		accountID: struct{}{},
+	}
+)
+
+// SetAccountID inserts the account ID value into context.
+func SetAccountID(ctx context.Context, value string) context.Context {
+	return keys.setAccountID(ctx, value)
+}
+
+// GetAccountID returns the account ID from the context.
+func GetAccountID(ctx context.Context) (string, bool) {
+	return keys.getAccountIDFromContext(ctx)
+}
+
+func (c contextKeys) getAccountIDFromContext(ctx context.Context) (string, bool) {
+	if nil != ctx {
+		accountID, ok := ctx.Value(keys.accountID).(string)
+		return accountID, ok
+	}
+
+	return "", false
+}
+
+func (c contextKeys) setAccountID(ctx context.Context, value string) context.Context {
+	if nil == ctx {
+		ctx = context.Background()
+	}
```

Comment:
I believe it's possible that the `accountID` struct underlying this could also be nil.  I don't see a test for the underlying `contextKey.Get/Set` functions for that case.

Reply:
I feel like accountID struct can't be nil. I get the type error `Cannot convert 'nil' to type 'struct{}'` when I attempt to add the following: 

```
	if nil == c.accountID {
		c.accountID = struct{}{}
	}
```

I think it could only be nil if the type `contextKeys.accountID` was a pointer like this:
```
type contextKeys struct {
	accountID *struct{}
}
```

Reply:
fair enough.